src/foam/nanos/crunch/UserCapabilityJunctionDAO.js

```diff
@@ -59,11 +43,13 @@ foam.CLASS({
       ],
       documentation: `Check if current user has permission to add this junction`,
       javaCode: `
-        User user = getUser(x);
-        User agent = ((Subject) x.get("subject")).getRealUser();
+        Subject subject = (Subject) x.get("subject");
+        User user = subject.getUser();
+        User realUser = subject.getRealUser();
+
         AuthService auth = (AuthService) x.get("auth");
-        boolean isOwner = obj.getSourceId() == user.getId() || ( agent != null && obj.getSourceId() == agent.getId() );
-        if ( ! isOwner && ! auth.check(x, "ucj.addPermission") ) throw new AuthorizationException(ERROR_MSG + agent.getId()+":"+user.getId());
+        boolean isOwner = obj.getSourceId() == user.getId() || obj.getSourceId() == realUser.getId();
+        if ( ! isOwner && ! auth.check(x, "*") ) throw new AuthorizationException();
```

Comment:
Don't think we should revert these changes - consider this https://github.com/foam-framework/foam2/pull/3597

think this is more correct: 'if ( ! isOwner && ! auth.check(x, "ucj.addPermission") ) throw new AuthorizationException(ERROR_MSG + agent.getId()+":"+user.getId());'

Reply:
huh must be a weird git thing, I was seeing "service.*" for the auth check locally so I changed to something that made more sense. will change, thanks